### PR TITLE
Onboarding maintainer checklist

### LINKED ISSUE #346: New maintainer induction checklist in Maintenance guide
Whenever a new package maintainer is added to the project, they will need to be informed of some procedures (protocols for merging PRs, review, releasing, etc) and be added to other accounts (CI, conda-forge, GitHub, PyPI, Google analytics, etc).

After discussion in #344, it’s clear that we need a checklist with these things to go over new maintainers when they join so we can make sure they are equipped with the proper permissions and information.

### PR DESCRIPTION
**Pull request by willschlitzer**
As discussed in #346, this is a pull request to gradually list the accesses and resources to use when becoming a new PyGMT maintainer.

Fixes #346.

### CODE REVIEW

**Comment by seisman** [comment]
Do you think we should put the checklist in the [Maintainers Guide](https://github.com/GenericMappingTools/pygmt/blob/master/MAINTENANCE.md). The reasons are:

1. We already have many files in the root directory of the repository. I prefer not to add more files.
2. Maintainers need to read the guide to know how the PyGMT project works. It may be easier for them to learn everything from one file.

File: MAINTENANCE.md [hunk 1]
```diff
@@ -6,6 +6,13 @@ making releases, creating packages, etc.
 If you want to make a contribution to the project, see the
 [Contributing Guide](CONTRIBUTING.md) instead.
 
+## Onboarding Access Checklist
+
+### Python-Maintainers
+Get added to the python-maintainers team on GitHub
+
+### Generic Mapping Tools organization
+Get added to the GenericMappingTools organization on GitHub
```
Comment (weiji14) [inline MAINTENANCE.md:15]:
```suggestion
- [ ] Added to [python-maintainers](https://github.com/orgs/GenericMappingTools/teams/python-maintainers) team in the [GenericMappingTools](https://github.com/orgs/GenericMappingTools/teams/) organization on GitHub (gives 'maintain' permissions)
- [ ] Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
- [ ] Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
- [ ] Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) (needed for handling releases)
```

Adding more stuff to the checklist 😄

**Comment by willschlitzer** [comment]
@seisman @weiji14 What else should I include on this (and/or get access to myself)?

File: MAINTENANCE.md [hunk 2]
```diff
@@ -6,6 +6,12 @@ making releases, creating packages, etc.
 If you want to make a contribution to the project, see the
 [Contributing Guide](CONTRIBUTING.md) instead.
 
+## Onboarding Access Checklist
+
+- [ ] Added to [python-maintainers](https://github.com/orgs/GenericMappingTools/teams/python-maintainers) team in the [GenericMappingTools](https://github.com/orgs/GenericMappingTools/teams/) organization on GitHub (gives 'maintain' permissions)
+- [ ] Added as moderator on [GMT forum](https://forum.generic-mapping-tools.org) (to see mod-only discussions)
+- [ ] Added as member on the [PyGMT devs Slack channel](https://pygmtdevs.slack.com) (for casual conversations)
+- [ ] Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) (needed for handling releases)
```
Comment (seisman) [inline MAINTENANCE.md:14]: The checklist looks good to me. 

> - [ ] Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) (needed for handling releases)

This step is usually unnecessary, but releases are handled by github actions automatically.

Comment (weiji14) [reply MAINTENANCE.md:14]: Maybe list hackmd as optional too.

Comment (seisman) [inline MAINTENANCE.md:14]:
```suggestion
- [ ] Added as maintainer on [PyPI](https://pypi.org/project/pygmt/) and [Test PyPI](https://test.pypi.org/project/pygmt) [optional]
- [ ] Added as member on [HackMD](https://hackmd.io/@pygmt) [optional]
```

Comment (weiji14) [reply MAINTENANCE.md:14]: Ping @willschlitzer to apply this change. I think we can merge this PR soon.

**Review by weiji14 — APPROVED** [review]

**Pull request merged**